DQMOffline/Lumi/src/ElectronIdentifier.cc

```diff
@@ -0,0 +1,188 @@
+   cuts_["DETAINSEED"]["MEDIUM"]["BARREL"] = 0.00311;
+   cuts_["DETAINSEED"]["TIGHT"] ["BARREL"] = 0.00308;
+
+   cuts_["DETAINSEED"]["VETO"]  ["ENDCAP"] = 0.00895;
+   cuts_["DETAINSEED"]["LOOSE"] ["ENDCAP"] = 0.00868;
+   cuts_["DETAINSEED"]["MEDIUM"]["ENDCAP"] = 0.00609;
+   cuts_["DETAINSEED"]["TIGHT"] ["ENDCAP"] = 0.00605;
+
+   cuts_["DPHIIN"]["VETO"]  ["BARREL"] = 0.228;
+   cuts_["DPHIIN"]["LOOSE"] ["BARREL"] = 	0.222;
+   cuts_["DPHIIN"]["MEDIUM"]["BARREL"] = 	0.103;
+   cuts_["DPHIIN"]["TIGHT"] ["BARREL"] = 	0.0816;
+
+   cuts_["DPHIIN"]["VETO"]  ["ENDCAP"] = 0.213;
+   cuts_["DPHIIN"]["LOOSE"] ["ENDCAP"] = 	0.213;
+   cuts_["DPHIIN"]["MEDIUM"]["ENDCAP"] = 	0.045;
+   cuts_["DPHIIN"]["TIGHT"] ["ENDCAP"] = 	0.0394;
+
+   cuts_["HOVERE"]["VETO"]  ["BARREL"] = 0.356;
+   cuts_["HOVERE"]["LOOSE"] ["BARREL"] = 	0.298;
+   cuts_["HOVERE"]["MEDIUM"]["BARREL"] = 0.253;
+   cuts_["HOVERE"]["TIGHT"] ["BARREL"] = 0.0414;
+
+   cuts_["HOVERE"]["VETO"]  ["ENDCAP"] = 0.211;
+   cuts_["HOVERE"]["LOOSE"] ["ENDCAP"] = 0.101;
+   cuts_["HOVERE"]["MEDIUM"]["ENDCAP"] = 0.0878;
+   cuts_["HOVERE"]["TIGHT"] ["ENDCAP"] = 0.0641;
+
+   cuts_["ISO"]["VETO"]  ["BARREL"] = 0.175;
+   cuts_["ISO"]["LOOSE"] ["BARREL"] = 0.0994;
+   cuts_["ISO"]["MEDIUM"]["BARREL"] = 	0.0695;
+   cuts_["ISO"]["TIGHT"] ["BARREL"] = 	0.0588;
+
+   cuts_["ISO"]["VETO"]  ["ENDCAP"] = 0.159;
+   cuts_["ISO"]["LOOSE"] ["ENDCAP"] = 	0.107;
+   cuts_["ISO"]["MEDIUM"]["ENDCAP"] = 	0.0821;
+   cuts_["ISO"]["TIGHT"] ["ENDCAP"] = 	0.0571;
+
+   cuts_["1OVERE"]["VETO"]  ["BARREL"] = 0.299;
+   cuts_["1OVERE"]["LOOSE"] ["BARREL"] = 0.241;
+   cuts_["1OVERE"]["MEDIUM"]["BARREL"] = 0.134;
+   cuts_["1OVERE"]["TIGHT"] ["BARREL"] = 0.0129;
+
+   cuts_["1OVERE"]["VETO"]  ["ENDCAP"] = 0.15;
+   cuts_["1OVERE"]["LOOSE"] ["ENDCAP"] = 0.14;
+   cuts_["1OVERE"]["MEDIUM"]["ENDCAP"] = 0.13;
+   cuts_["1OVERE"]["TIGHT"] ["ENDCAP"] = 0.0129;
+
+   cuts_["MISSINGHITS"]["VETO"]  ["BARREL"] = 2;
+   cuts_["MISSINGHITS"]["LOOSE"] ["BARREL"] = 1;
+   cuts_["MISSINGHITS"]["MEDIUM"]["BARREL"] = 1;
+   cuts_["MISSINGHITS"]["TIGHT"] ["BARREL"] = 1;
+
+   cuts_["MISSINGHITS"]["VETO"]  ["ENDCAP"] = 3;
+   cuts_["MISSINGHITS"]["LOOSE"] ["ENDCAP"] = 1;
+   cuts_["MISSINGHITS"]["MEDIUM"]["ENDCAP"] = 1;
+   cuts_["MISSINGHITS"]["TIGHT"] ["ENDCAP"] = 1;
+
+   cuts_["CONVERSION"]["VETO"]  ["BARREL"] = 1;
+   cuts_["CONVERSION"]["LOOSE"] ["BARREL"] = 1;
+   cuts_["CONVERSION"]["MEDIUM"]["BARREL"] = 1;
+   cuts_["CONVERSION"]["TIGHT"] ["BARREL"] = 1;
+
+   cuts_["CONVERSION"]["VETO"]  ["ENDCAP"] = 1;
+   cuts_["CONVERSION"]["LOOSE"] ["ENDCAP"] = 1;
+   cuts_["CONVERSION"]["MEDIUM"]["ENDCAP"] = 1;
+   cuts_["CONVERSION"]["TIGHT"] ["ENDCAP"] = 1;
+}
+
+void ElectronIdentifier::setRho(double rho) {
+   if(rho >= 0) {
+      rho_ = rho;
+   } else {
+      throw;
+   }
+}
+void ElectronIdentifier::setID(std::string ID) {
+   bool is_available = true;
+   for (auto const cutmap : cuts_) {
```

Comment:
auto const & should be ok

Reply:
This part of the code is now removed since it was only relevant for the old, map-based cut implementation.